source/internal.d.ts

```diff
@@ -328,3 +356,83 @@ IsPrimitive<Object>
 ```
 */
 export type IsPrimitive<T> = [T] extends [Primitive] ? true : false;
+
+/**
+Returns the static, fixed-length portion of the given array, excluding variable-length parts.
+
+@example
+```
+type A = [string, number, boolean, ...string[]];
+type B = StaticPartOfArray<A>;
+//=> [string, number, boolean]
+```
+*/
+export type StaticPartOfArray<T extends UnknownArray, Result extends UnknownArray = []> =
+	T extends unknown
+		? number extends T['length'] ?
+			T extends readonly [infer U, ...infer V]
+				? StaticPartOfArray<V, [...Result, U]>
+				: Result
+			: T
+		: never; // Should never happen
+
+/**
+Returns the non-fixed-indexed part of the given array.
+
+@example
+```
+type A = [string, number, boolean, ...string[]];
+type B = NonFixedPartOfArray<A>;
+//=> string[]
+```
+*/
+export type NonFixedPartOfArray<T extends UnknownArray> =
```

Comment:
so according to the above, maybe `VariablePartOfArray` better? @sindresorhus

Reply:
👍